coupling/delwaq/generate.py

```diff
@@ -0,0 +1,359 @@
+"""Setup a Delwaq model from a Ribasim model and results."""
+
+import csv
+import shutil
+from datetime import timedelta
+from pathlib import Path
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+import ribasim
+from jinja2 import Environment, FileSystemLoader
+from util import (
+    strfdelta,
+    ugrid,
+    write_flows,
+    write_pointer,
+    write_volumes,
+)
+
+delwaq_dir = Path(__file__).parent
+
+env = Environment(autoescape=True, loader=FileSystemLoader(delwaq_dir / "template"))
+
+# Add evaporation edges, so mass balance is correct
+# To simulate salt increase due to evaporation, set to False
+USE_EVAP = True
+
+
+def generate(modelfn: Path) -> tuple[nx.DiGraph, set[str]]:
+    """Generate a Delwaq model from a Ribasim model and results."""
+
+    # Read in model and results
+    model = ribasim.Model.read(modelfn)
+    basins = pd.read_feather(modelfn.parent / "results" / "basin.arrow")
+    flows = pd.read_feather(modelfn.parent / "results" / "flow.arrow")
+
+    output_folder = delwaq_dir / "model"
+    output_folder.mkdir(exist_ok=True)
+
+    # Simplify network, only keep Basins and Boundaries
+    G = nx.DiGraph()
+    for row in model.node_table().df.itertuples():
+        if "Control" not in row.node_type:
```

Comment:
Would using something like this be more robust?

https://github.com/Deltares/Ribasim/blob/333e2acc23a7789b3f35f469ff3920a3a226dddd/python/ribasim/ribasim/geometry/edge.py#L60-L62

That would also include LevelDemand.